sdk/workloads/Azure.ResourceManager.Workloads/tests/Tests/SviCRUDTests.cs

```diff
@@ -0,0 +1,268 @@
+            deploymentConfiguration.SoftwareConfiguration = softwareConfiguration;
+
+            SapVirtualInstanceData testSapWorkload = new SapVirtualInstanceData(
+                location: AzureLocation.EastUS2,
+                environment: SapEnvironmentType.NonProd,
+                sapProduct: SapProductType.S4Hana,
+                configuration: deploymentConfiguration);
+
+            return testSapWorkload;
+        }
+
+        /// <summary>
+        /// Gets the secret from key vault.
+        /// </summary>
+        /// <param name="secretName">The secret name for install.</param>
+        /// <returns>The secret value.</returns>
+        private string GetKeyVaultSecret(string secretName)
+        {
+            var kvUri = "https://waas-service-ct-kv.vault.azure.net/";
+            var certificate = GetCertificateBySubjectName(
+                        "HRME2ANew");
+
+            var client = new SecretClient(new Uri(kvUri), new ClientCertificateCredential(
+                "72f988bf-86f1-41af-91ab-2d7cd011db47",
+                "d4b3c6a3-2fd1-4f46-b0c1-37220ff8d54d",
+                certificate));
+
+            KeyVaultSecret secret = client.GetSecret(secretName);
+            return secret.Value;
+        }
+
+        /// <summary>
+        /// Gets the certificate from local store.
+        /// </summary>
+        /// <param name="certSubjectName">Certificate subject name.</param>
+        /// <returns>Certificate object.</returns>
+        public static X509Certificate2 GetCertificateBySubjectName(string certSubjectName)
+        {
+            using (var machineCertStore = new X509Store(StoreName.My, StoreLocation.LocalMachine))
+            {
+                machineCertStore.Open(OpenFlags.ReadOnly);
+
+                var certCollection = machineCertStore.Certificates.Find(
+                    X509FindType.FindBySubjectName,
+                    certSubjectName,
+                    false);
+                if (certCollection.Count == 0)
+                {
+                    throw new ArgumentException(
+                        $"Certificate with subject name: {certSubjectName} is not present in local " +
+                        $"machine store.");
+                }
+                else
+                {
+                    X509Certificate2 selectedCert = certCollection[0];
+                    foreach (X509Certificate2 cert in certCollection)
+                    {
+                        if (cert.NotBefore > selectedCert.NotBefore)
+                        {
+                            selectedCert = cert;
+                        }
+                    }
+
+                    return selectedCert;
+                }
+            }
+        }
+
+        [System.Diagnostics.CodeAnalysis.SuppressMessage(
+    "StyleCop.CSharp.DocumentationRules",
+    "SA1600:Elements should be documented",
+    Justification = "keeping related things together")]
+        public class OsImage
```

Comment:
why is this required? are we overriding default SDK model behaviour?

Reply:
Removed